rust/src/connection/network/transmitter/rpc.rs

```diff
@@ -151,7 +153,28 @@ impl RPCTransmitter {
             Request::Transaction(transaction_request) => {
                 let req = transaction_request.into_proto();
                 let req_id = RequestID::from(req.req_id.clone());
-                let (request_sink, response_source) = rpc.transaction(req).await?;
+                let (request_sink, mut response_source) = rpc.transaction(req).await?;
+                match response_source.next().await {
```

Comment:
Maybe it worth making a separate function just to put a lable "initial response" on it